src/main/java/com/autonomic/tmc/auth/ClientCredentialsTokenSupplier.java

```diff
@@ -19,43 +19,27 @@
  */
 package com.autonomic.tmc.auth;
 
-import com.nimbusds.oauth2.sdk.AccessTokenResponse;
-import com.nimbusds.oauth2.sdk.AuthorizationGrant;
-import com.nimbusds.oauth2.sdk.ClientCredentialsGrant;
-import com.nimbusds.oauth2.sdk.ParseException;
-import com.nimbusds.oauth2.sdk.TokenRequest;
+import com.autonomic.tmc.auth.exception.BaseSdkException;
+import com.autonomic.tmc.auth.exception.SdkClientException;
+import com.autonomic.tmc.auth.exception.SdkServiceException;
 import com.nimbusds.oauth2.sdk.TokenResponse;
-import com.nimbusds.oauth2.sdk.auth.ClientAuthentication;
-import com.nimbusds.oauth2.sdk.auth.ClientSecretPost;
-import com.nimbusds.oauth2.sdk.auth.Secret;
-import com.nimbusds.oauth2.sdk.http.HTTPResponse;
-import com.nimbusds.oauth2.sdk.id.ClientID;
 import com.nimbusds.oauth2.sdk.token.AccessToken;
-import java.io.IOException;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.time.Instant;
-import java.time.temporal.ChronoUnit;
 import lombok.Builder;
 import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
 
 /**
- * This class provides an OAuth 2.0 - Client Credentials Grant based implementation of {@link TokenSupplier}.
+ * This class provides an OAuth 2.0 - Client Credentials Grant based implementation of {@link
+ * TokenSupplier}.
  *
- * <p>Reference: <a target="_blank" href="https://tools.ietf.org/html/rfc6749#section-1.3.4">RFC 6749 section-1.3.4</a></p>
+ * <p>Reference: <a target="_blank" href="https://tools.ietf.org/html/rfc6749#section-1.3.4">RFC
+ * 6749 section-1.3.4</a></p>
  */
+@Slf4j
 public class ClientCredentialsTokenSupplier implements TokenSupplier {
 
-    private static final String DEFAULT_TOKEN_URL = "https://accounts.autonomic.ai/auth/realms/iam/protocol/openid-connect/token";
-
-    // properties
-    private final String clientId;
-    private final String clientSecret;
-    private final String tokenUrl;
-    private final URI tokenEndpoint;
-
-    // state
-    private Token token = null;
+    private final BaseClientCredentialsTokenSupplier baseSupplier;
+    private Token token;
```

Comment:
Why do we need a token here if BaseClientCredentialsTokenSupplier has a token, too?